.github/workflows/ci.yml

```diff
@@ -16,9 +24,36 @@ on:
 env:
   REGISTRY: ghcr.io
   IMAGE_NAME: ${{ github.repository }}
+  TEST_PROJECT_SYNID: syn7208886
 
 jobs:
 
+  determine-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      non-test-changes: ${{ steps.get-changes.outputs.non-test-changes }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Determine Changes
+        id: get-changes
+        run: |
+          if [ "${{ github.event_name }}" == "push" ]; then
+            echo "Handling a push event..."
+            ALL_CHANGED_FILES=$(git diff --name-only HEAD^ HEAD)
+          elif [ "${{ github.event_name }}" == "pull_request" ]; then
+            echo "Handling a pull request event..."
+            git fetch origin ${{ github.base_ref }}
+            ALL_CHANGED_FILES=$(git diff --name-only origin/${{ github.base_ref }} HEAD)
+          fi
+          echo "All Changed Files: $ALL_CHANGED_FILES"
+          NON_TEST_CHANGES=$(echo "$ALL_CHANGED_FILES" | grep -v '^tests/' || true)
+          echo "Non-Test Changes: $NON_TEST_CHANGES"
+          echo "::set-output name=non-test-changes::$NON_TEST_CHANGES"
```

Comment:
Just to get an understanding here. are you only running these tests when there are changes outside of the test folder?

Reply:
Yup, I didn't want the integration tests to run when there are only changes to the unit tests

Reply:
How would you differentiate between integration and unit tests being added?

Reply:
Hmmm our integration tests are going to be here as jobs in the github workflow while our unit tests are in the `tests` folder? That would be the differentiation

Reply:
I don't think so because we want the `test` GH action job to run the unit tests in that folder

Reply:
You're right, I forgot the path exclude is for the entire workflow.